src/Simulation/Simulators/Microsoft.Quantum.Simulators.csproj

```diff
@@ -5,7 +5,7 @@
   <Import Project="..\Common\Simulators.Dev.props" />
 
   <PropertyGroup>
-    <LanguageVersion>8.0</LanguageVersion>
+    <LanguageVersion>10.0</LanguageVersion>
```

Comment:
I believe this change affects all the simulators, not only the OpenSystemsSimulator.
The CI runs the tests for Release config only.
Just in case I would recommend to build locally the Debug config of the simulators and run the Debug config tests. Something like 
```powershell
dotnet clean Simulation.sln
dotnet build Simulation.sln -c Debug
dotnet test Simulation.sln -c Debug

# Or even more reliable:
$Env:BUILD_CONFIGURATION="Debug"
pwsh ./bootstrap.ps1
dotnet test Simulation.sln
```

Reply:
Sounds good, I've not run into any issues so far running locally, as C# 10 is back-compat with 8, but I agree with being cautious here.